schemas/when_rule/definitions.json

```diff
@@ -0,0 +1,87 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema",
+  "when": {
+    "type": "array",
+    "items": {
+      "anyOf": [
+        {
+          "$ref": "compare_to_value.json"
+        },
+        {
+          "$ref": "compare_to_list.json"
+        },
+        {
+          "$ref": "compare_to_answer_store.json"
+        }
+      ]
+    },
+    "minItems": 1
+  },
+  "answer_identifier": {
+    "type": "string",
+    "description": "The id of an answer from which to obtain the value"
+  },
+  "metadata_identifier": {
+    "type": "string",
+    "description": "The id of a metadata provided by the calling service from which to obtain the value"
+  },
+  "comparison_value": {
+    "type": [
+      "string",
+      "boolean",
+      "integer"
+    ],
+    "description": "The value to compare against"
+  },
+  "comparison_values": {
+    "type": "array",
+    "description": "An array of values to compare against",
+    "minItems": 1
+  },
+  "comparison_identifier": {
+    "type": "string",
+    "description": "The id of an answer from which to obtain the value and compared to"
+  },
+  "date_comparison_value": {
+    "type": "object",
+    "properties": {
+      "id": {
+        "$ref": "#/answer_identifier"
+      },
+      "meta": {
+        "$ref": "#/metadata_identifier"
+      },
+      "value": {
+        "allOf": [
+          {
+            "$ref": "#/comparison_value"
+          },
+          {
+            "pattern": "^(now|\\d{4}\\-(0?[1-9]|1[012])(|\\-(0?[1-9]|[12][0-9]|3[01])))$"
```

Comment:
There is already some regex used for dates in `date_value` in the string interpolation definitions schema. We should consolidate these if appropriate.

Reply:
I've reused the `date_value` and seem to work fine, although this pattern doesn't seem to match the same.